File: packages/backend/src/game/game.service.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { UUID } from 'crypto';
import { CreateGameDto } from 'src/game/dto/create-game.dto';
import { GameRoundModel } from 'src/game/model/game-round.model';
import { GameModel } from 'src/game/model/game.model';
import { Repository } from 'typeorm';

@Injectable()
export class GameService {
  constructor(
    @InjectRepository(GameModel)
    private readonly gameRepository: Repository<GameModel>,
    @InjectRepository(GameRoundModel)
    private readonly gameRoundRepository: Repository<GameRoundModel>,
  ) {}

  async createGame(data: CreateGameDto): Promise<UUID> {
    try {
      return await this.gameRepository.manager.transaction(async (manager) => {
        const game = manager.create(GameModel, { name: data.name });
        const savedGame = await manager.save(game);

        const rounds = data.rounds.map((round) =>
          manager.create(GameRoundModel, {
            ...round,
            game: savedGame,
          }),
        );
        await manager.save(rounds);

        return savedGame.id;
      });
    } catch (error) {
      // TODO
      console.error('Error creating game:', error);
      throw error; // Rethrow the error to be handled by the controller
    }
  }
}

File: packages/backend/src/features/game/service/GameService.kt
package quizzem.backend.features.game.service

import org.springframework.stereotype.Service
import quizzem.backend.features.game.api.dto.CreateGameDto
import quizzem.backend.features.game.model.GameModel
import quizzem.backend.features.game.repository.GameRepository
import quizzem.backend.features.question.repository.QuestionRepository
import java.util.*

@Service
class GameService(
    private val gameRepository: GameRepository,
    private val questionRepository: QuestionRepository,
) {
    fun listGames(): List<GameModel> {
        return gameRepository.findAll()
    }

    fun createGame(dto: CreateGameDto): UUID {
        val saved = gameRepository.save(GameModel(title = dto.title, description = dto.description))
        return saved.id
    }
}